pkg/action/executor/k8s_controller.go

```diff
@@ -153,6 +155,42 @@ func (c *parentController) update(updatedSpec *k8sControllerSpec) error {
 	return err
 }
 
+func (c *parentController) waitForReplicaSetToBeScheduled(name string) (bool, error) {
+	if err := wait.Poll(DefaultRetrySleepInterval, DefaultRetryTimeout, func() (bool, error) {
+		obj, errInternal := c.clients.dynNamespacedClient.Get(context.TODO(), name, metav1.GetOptions{})
+		if errInternal != nil {
+			return false, errInternal
+		}
+
+		var replicas, updatedReplicas int64
+		found := false
+
+		replicas, found, errInternal = unstructured.NestedInt64(obj.Object, "spec", "replicas")
+		if errInternal != nil || !found {
+			return false, errInternal
+		}
+
+		if obj.GetKind() == util.KindDaemonSet {
+			updatedReplicas, found, errInternal = unstructured.NestedInt64(obj.Object, "status", "updatedNumberScheduled")
+		} else {
+			updatedReplicas, found, errInternal = unstructured.NestedInt64(obj.Object, "status", "updatedReplicas")
+		}
```

Comment:
`updatedReplicas`  indicates how many replicas has already been scheduled by the scheduler, usually it's updated after the new replicas go to the `creating` phase.  At that phase, the pod has been scheduled and created without any quota issue. I think it should be ok.  we can use `ReadyReplicas `, but that means it will take more time for our waiting. and I'm not sure if `ReadyReplicas` depends on `readiness` or `livessness` probe or not, if it does, it might be associated with the user's business logic.

Reply:
Even if there is no quota issue, could there be issue of cluster capacity, in other words, could the pod be stuck in `Pending` state? If this is a possibility, we should confirm with Eva if this is acceptable.

Reply:
@evat-pm  Could you please help confirm Meng's concern about which field should be used to check to determine the resizing action has been successfully executed.